packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -131,21 +149,33 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
       service: 'codepipeline',
       resource: this.pipelineName
     });
+
+    for (const stage of props.stages || []) {
+      this.addStage(stage);
+    }
   }
 
   /**
-   * Convenience method for creating a new {@link Stage},
-   * and adding it to this Pipeline.
+   * Adds a Stage to this Pipeline.
    *
-   * @param name the name of the newly created Stage
-   * @param props the optional construction properties of the new Stage
-   * @returns the newly created Stage
+   * @param stage the newly created Stage to add to this Pipeline
+   * @param props optional Stage insertion properties
    */
-  public addStage(name: string, props?: CommonStageProps): Stage {
-    return new Stage(this, name, {
-      pipeline: this,
-      ...props,
-    });
+  public addStage(stage: cpapi.IStage, props: StageInsertionProps = {}): Pipeline {
```

Comment:
Why do we even need a public concrete `Stage` class? What's the use case for this? Wouldn't it make more sense to always create stages in the context of a pipeline?

So if I want to get a stage handle, I'll just do:

```ts
const stage = pipeline.addStage('Source');
```

And then, I can add actions to it:

```ts
stage.addAction(myAction);
```

Reply:
Changed. `Stage` is no longer exported from the `aws-codepipeline` package.